src/gitlab.ts

```diff
@@ -137,16 +137,77 @@ export const runCommandInGitlabPipeline = async (ctx: Context, task: Task) => {
   */
   await cmdRunner.run("git", ["push", "--force", gitlabRemote, "HEAD"])
 
+  /*
+    Wait until the branch is actually present on GitLab after pushing it. We've
+    noted this measure is required in
+    https://github.com/paritytech/polkadot/pull/5524#issuecomment-1128029579
+    because the pipeline creation request was sent too soon, before GitLab
+    registered the branch, therefore causing the "Reference not found" message.
+  */
+  let wasBranchRegistered = false
+  const waitForBranchMaxTries = 3
+  const waitForBranchRetryDelay = 1024
+
+  const branchPresenceUrl = `https://${
+    gitlab.domain
+  }/api/v4/projects/${encodeURIComponent(
+    gitlabProjectPath,
+  )}/repository/branches/${encodeURIComponent(branchName)}`
+  for (
+    let waitForBranchTryCount = 0;
+    waitForBranchTryCount < waitForBranchMaxTries;
+    waitForBranchTryCount++
+  ) {
+    logger.info(
+      branchPresenceUrl,
+      `Sending request to see if the branch for task ${task.id} is ready`,
+    )
+    const response = await fetch(branchPresenceUrl, {
+      headers: { "PRIVATE-TOKEN": gitlab.accessToken },
+    })
+    if (
+      // The branch was not yet registered on GitLab; wait for it...
+      response.status === 404
+    ) {
+      logger.info(
+        `Branch of task ${task.id} was not found. Waiting before retrying...`,
+      )
+      await millisecondsDelay(waitForBranchRetryDelay)
+    } else if (response.ok) {
+      wasBranchRegistered = true
+      break
+    } else {
+      throw new Error(
+        `Request to ${branchPresenceUrl} failed: ${await response.text()}`,
+      )
+    }
+  }
+
+  if (!wasBranchRegistered) {
+    throw new Error(
+      `Task's branch was not registered on GitLab after ${
+        waitForBranchMaxTries * waitForBranchRetryDelay
+      }ms`,
+    )
+  }
+
+  const pipelineCreationUrl = `https://${
+    gitlab.domain
+  }/api/v4/projects/${encodeURIComponent(
+    gitlabProjectPath,
```

Comment:
That seems unnecessarily verbose and excessive. It would turn this single variable into three separate variables + three lines of code. Specially given that those variables are only used once, I don't see any reason to turn them into constants. Do you think this suggestion improves readability significantly?

Reply:
https://github.com/paritytech/command-bot/pull/67/commits/a8fc8476429cf5fefd78a0b2db33c22b69aec996